src/parser/ast.c

```diff
@@ -303,6 +285,10 @@ AST_Validation AST_Validate(const AST_Query *ast, char **reason) {
     return AST_INVALID;
   }
 
+  if(_Aliases_Defined(ast, reason) != AST_VALID) {
+    return AST_INVALID;
+  }
+
   if (_Validate_MATCH_Clause(ast, reason) != AST_VALID) {
     return AST_INVALID;
   }
```

Comment:
We can't yet have expressions like `MATCH (a) UNWIND [a.x]`, but we can still have `UNWIND [a.x]`, which will produce an assertion failure when we try to evaluate the arithmetic expression.

We should add a _Validate_UNWIND_Clause` to make sure our arithmetic expressions are sensible.